lib/Utils/PlatformPosix.cpp

```diff
@@ -0,0 +1,201 @@
+//--------------------------------------------------------------------*- C++ -*-
+// CLING - the C++ LLVM-based InterpreterG :)
+// author:
+//
+// This file is dual-licensed: you can choose to license it under the University
+// of Illinois Open Source License or the GNU Lesser General Public License. See
+// LICENSE.TXT for details.
+//------------------------------------------------------------------------------
+
+#include "cling/Utils/Platform.h"
+
+#if defined(LLVM_ON_UNIX)
+
+#include "cling/Utils/Paths.h"
+
+#include <string>
+#include <dlfcn.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <unistd.h>
+
+#ifdef __APPLE__
+ #include <sys/syslimits.h> // PATH_MAX
+#else
+ #include <array>
+ #include <atomic>
+ #include <limits.h>
+#endif
+
+#define PATH_MAXC (PATH_MAX+1)
+
+namespace cling {
+namespace utils {
+namespace platform {
+
+std::string GetCwd() {
+  char Buffer[PATH_MAXC];
+  if (::getcwd(Buffer, sizeof(Buffer)))
+    return Buffer;
+
+  ::perror("Could not get current working directory");
+  return std::string();
+}
+
+static void DLErr(std::string* Err) {
+  if (!Err)
+    return;
+  if (const char* DyLibError = ::dlerror())
```

Comment:
Largely I don't really care, but my point was this is now user facing code, so it can be used in the Interpreter:
[cling]: void* lib = DLOpen();
[cling]: more commands, and lib is discovered to be null.
[cling]: dlerror() // show the error that was generated.